src/components/Article.js

```diff
@@ -1,22 +1,83 @@
 import styles from "./article.module.scss";
+import { Link, useSearchParams } from "react-router-dom";
+import starImage from "images/star-svgrepo-com.svg";
+import inactiveStarImage from "../images/star-inactive.com.svg";
+import { useState } from "react";
 
-export default function Articles({ article }) {
+export default function Articles({ article, articleIndex, pageNumber }) {
+  let currentIndexInStorage;
   const date = article.publishedAt.slice(0, 10);
+  const [searchParams] = useSearchParams();
+  const [buttonImage, setButtonImage] = useState(
+    JSON.parse(localStorage.getItem("favorites")).some(isAlreadyInStorage)
+      ? starImage
+      : inactiveStarImage
+  );
+
+  function isAlreadyInStorage(element, index) {
+    currentIndexInStorage = index;
+    return element.title === article.title;
+  }
 
   return (
     <div className={styles["article"]}>
-      <h2 className={styles["article__title"]}>{article.title}</h2>
-      <div className={styles["article__content"]}>
+      <button
+        className={styles["article__favorite-button"]}
+        onClick={() => {
+          if (!localStorage.getItem("favorites")) {
+            const favorites = [];
+            localStorage.setItem("favorites", JSON.stringify(favorites));
+          }
+
+          const favoritesStorage = JSON.parse(
+            localStorage.getItem("favorites")
+          );
+
+          if (favoritesStorage.some(isAlreadyInStorage)) {
+            favoritesStorage.splice(currentIndexInStorage, 1);
+            localStorage.setItem("favorites", JSON.stringify(favoritesStorage));
```

Comment:
Please consider puting it after the condition because you do it no matter true or false in the condition